File: 0001-two-sum/0001-two-sum.cpp
#include <unordered_map>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> checked;
        for(int i = 0; i < nums.size(); i++){
            int need = target - nums[i];
            if(checked.find(need) == checked.end()){
                checked[nums[i]] = i;
            }else{
                return vector<int>{i, checked[need]};
            }
        }
        return vector<int>{};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        map_nums = {}

        for i in range(len(nums)):
            if nums[i] in map_nums:
                return [i, map_nums[nums[i]]]
            subs = target - nums[i]
            map_nums[subs] = i